### ISSUE TITLE
Joplin stuck on syncing the encryption MasterKey

### ISSUE THREAD

**Issue by dev4223** [laurent22/joplin#804]
### Operating system

- Windows 10

### Application

- Desktop

Hello,

I came from Evernote an get all my notes (some ~600) into Joplin desktop app under Windows 10. I exported all my notes from Evernote and imported them into Joplin. The I do some cleaning on my tags and notes. Syncing with Nextcloud then takes some time but finished without complications.

Then I wanted to use the E2EE feature and filled in an encryption phrase. Encryption worked and then the sync did also work for all the tags and notes, but not for the MasterKey. The sync process stuck there and the MasterKey did not get synced.

**Comment by laurent22** [comment]
How do you know the master key didn't get synced?

**Comment by dev4223** [comment]
Because of the sync status screen. I did a fresh sync today following your E2EE Guide. But it hangs during sync again. Please see the attached screenshots.
![180922-joplin-hangs-on-encrypted-sync](https://user-images.githubusercontent.com/400822/45916111-dedd8f00-be60-11e8-9681-71a7d0f5134c.jpg)
![180922-joplin-hangs-on-encrypted-sync-status](https://user-images.githubusercontent.com/400822/45916112-e0a75280-be60-11e8-9ad6-032410cac9bb.jpg)

**Comment by dev4223** [comment]
sorry, hi the wrong button

**Comment by dev4223** [comment]
The status always is "Working" and it never changes back. Now it says, it synced just one item but it stays on "Working" . And because I imported all my notes in the beginning from an .jex file and synced them, I know approximately how long it took. I've done this for testing 3 or 4 times now. And on the sync status screen the master key always is on not synced.
Yesterday night after the syncing I let my laptop runing until now. So the encrypted syncing should be ready now. It's twice or three times the time it took for the first sync.

**Comment by dev4223** [comment]
ok, now I managed to get a full sync working. Firstly I cleaned up my notes, which I've imported from Evernote, looking for large attachements. I deleted all attachments over 4 MB. There was one note with 8 to 10 animated GIFs with around 8 MB each. I think, this was the largest note by overall size of the attachments.
After this cleaning the sync was working.
So to answer your question: I thought the master key wasn't synced because of the Joplin status screen entry for MasterKey, which said 0/1 (Synced Objects/All Objects). 
So for the stuck-on-sync-problem this is not really a solution, because I don't know whether it comes from deleting large attachments or because I've edited these notes with large attachments.
